x-pack/plugins/security_solution/server/lib/detection_engine/signals/search_after_bulk_create.ts

```diff
@@ -96,116 +104,132 @@ export const searchAfterAndBulkCreate = async ({
     we only want 500. So maxResults will help us control how
     many times we perform a search_after
   */
-  let maxResults = ruleParams.maxSignals;
 
-  // Get
+  const totalToFromTuples = getSignalTimeTuples({
+    logger,
+    ruleParamsFrom: ruleParams.from,
+    ruleParamsTo: ruleParams.to,
+    ruleParamsMaxSignals: ruleParams.maxSignals,
+    gap,
+    previousStartedAt,
+    interval,
+  });
+  const useSortIds = totalToFromTuples.length <= 1;
+  logger.debug(`totalToFromTuples: ${totalToFromTuples.length}`);
+  while (totalToFromTuples.length > 0) {
+    const tuple = totalToFromTuples.pop();
+    if (tuple == null || tuple.to == null || tuple.from == null) {
+      logger.error(`[-] malformed date tuple`);
+      toReturn.success = false;
+      return toReturn;
+    }
+    searchResultSize = 0;
+    while (searchResultSize < tuple.maxSignals) {
+      try {
+        logger.debug(`sortIds: ${sortId}`);
+        const {
+          // @ts-ignore https://github.com/microsoft/TypeScript/issues/35546
+          searchResult,
+          searchDuration,
+        }: { searchResult: SignalSearchResponse; searchDuration: string } = await singleSearchAfter(
+          {
+            // @ts-ignore we are using sortId before being assigned but that's ok.
```

Comment:
mm yeah this might have been leftover from when I was testing things. It was complaining I was using sortId before assignment but that is not the case anymore. thanks!